### PR TITLE
Move implementation of WellOperators out of BlackoilWellModel

### PR DESCRIPTION
**Pull request by akva2**
We already have adapters for this, the additional indirection is seems unnecessary to me. As only const access is required, the additional accessors seems like a small price to pay.

### CODE REVIEW

**Comment by akva2** [comment]
jenkins build this please

File: opm/simulators/linalg/WellOperators.hpp [hunk 1]
```diff
@@ -84,14 +84,43 @@ class WellModelAsLinearOperator : public LinearOperatorExtra<X, Y>
     void apply(const X& x, Y& y) const override
     {
         OPM_TIMEBLOCK(apply);
-        wellMod_.apply(x, y);
+        for (const auto& well : this->wellMod_) {
+            // Well equations B and C uses only the perforated cells, so need to apply on local vectors
```
Comment (atgeirr) [inline opm/simulators/linalg/WellOperators.hpp:88]: I think we can reduce duplication in a good way by moving the content of this loop to a separate function `applySingleWell()`, then the apply() function of both classes here would not duplicate each other.

File: opm/simulators/linalg/WellOperators.hpp [hunk 2]
```diff
@@ -125,6 +154,10 @@ class WellModelAsLinearOperator : public LinearOperatorExtra<X, Y>
 
 protected:
     const WellModel& wellMod_;
+
+    mutable X x_local_{};
```
Comment (atgeirr) [inline opm/simulators/linalg/WellOperators.hpp:158]: I think `mutable` variables should always be explained in a comment. (Can be lifted from the BlackoilWellModel class.)

**Review by atgeirr — COMMENTED** [review]
I think this is a good idea, and it should be merged. I believe this is a useful step also if/when we move further toward separating the well-related linear algebra from the well modeling classes.

File: opm/simulators/linalg/WellOperators.hpp (hunk 1 above)
Comment (akva2) [reply opm/simulators/linalg/WellOperators.hpp]: good idea. i have added a commit that does this. furthermore, applyscaleadd is also identical so i decided to share them.

File: opm/simulators/linalg/WellOperators.hpp (hunk 2 above)
Comment (akva2) [reply opm/simulators/linalg/WellOperators.hpp:169]: alright, lifted the comment.

**Comment by akva2** [comment]
jenkins build this please

**Review by atgeirr — APPROVED** [review]
Great simplifications, now the difference stands out very clearly between the Domain and global version!

**Pull request merged**